pageserver/src/page_cache.rs

```diff
@@ -758,8 +867,11 @@ impl PageCache {
     /// Find a slot to evict.
     ///
     /// On return, the slot is empty and write-locked.
-    fn find_victim(&self) -> anyhow::Result<(usize, tokio::sync::RwLockWriteGuard<SlotInner>)> {
-        let iter_limit = self.slots.len() * 10;
+    async fn find_victim(
+        &self,
+        _permit_witness: &PinnedSlotsPermit,
+    ) -> anyhow::Result<(usize, tokio::sync::RwLockWriteGuard<SlotInner>)> {
+        let iter_limit = self.slots.len() * (usize::try_from(MAX_USAGE_COUNT).unwrap());
```

Comment:
Objections to this change here, given my [theories about infalliblity](https://github.com/neondatabase/neon/pull/5319#discussion_r1335765519) have been [invalidated](https://github.com/neondatabase/neon/pull/5319#discussion_r1335822762)?

Reply:
we fail here more quickly, as MAX_USAGE_COUNT is 5 instead of the old 10. I think it's fine. In the end, it's an arbitrary limit and this PR has, I believe, greatly reduced the chances of this occurring. So I'd say it's fine, although one should put a comment that this limit is not derived from a hard invariant that if it is broken it's a bug, but a rather unlikely circumstance. The presence of the constant compared to a raw number makes it easier to think the prior. it does make sense to link the limit to MAX_USAGE_COUNT though.